src/EditorFeatures/Test/PasteTracking/PasteTrackingServiceTests.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.PasteTracking
+{
+    [UseExportProvider]
+    public class PasteTrackingServiceTests
+    {
+        [WpfFact]
+        [Trait(Traits.Feature, Traits.Features.RenameTracking)]
+        public void PasteTracking_MissingTextSpan_WhenNothingPasted()
+        {
+            var code = @"
+class C
+{
+$$
+}";
+
+            using (var testState = PasteTrackingTestState.Create(code, LanguageNames.CSharp))
+            {
+                testState.AssertMissingPastedTextSpan();
+            }
+        }
+
+        [WpfFact]
+        [Trait(Traits.Feature, Traits.Features.RenameTracking)]
+        public void PasteTracking_HasTextSpan_AfterPaste()
+        {
+            var code = @"
+class C
+{
+$$
+}";
+
+            var pasted = @"
+public void Main(string[] args)
+{
+}
+";
+
+            using (var testState = PasteTrackingTestState.Create(code, LanguageNames.CSharp))
+            {
+                var start = testState.HostDocument.CursorPosition.GetValueOrDefault();
+                var length = pasted.Length;
+                var expectedTextSpan = new Text.TextSpan(start, length);
+
+                testState.SendPaste(pasted);
```

Comment:
oh good.  we have these tests.  Honestly, i would be ok if these were hte *only* tests.